### ISSUE TITLE
URLs to external services should always be configuration, not code.

### ISSUE THREAD

**Issue by mwoodiupui** [DSpace/DSpace#3287]
There are several URLs for "foreign" services which are hard-coded in DSpace.  For example:  the stem of URLs for the DOI resolver is a `public static final String` at `org.dspace.identifier.DOI#RESOLVER`.  Even though these are supposed to be stable, experience shows that they do change:  `https://doi.org/` is now recommended in place of `http://dx.doi.org/`.  Having these wired into the code makes changing them more difficult than necessary.

We should make a list of these cases:
- [ ] org.dspace.identifier.DOI#RESOLVER
- [ ] org.dspace.handle.HandleServiceImpl#handlePrefix
- [ ] org.dspace.app.sherpa.SherpaService: "https://v2.sherpa.ac.uk/cgi/retrieve"
- [ ] org.dspace.google.GoogleRecorderEventListener#GoogleURL
* more to come 

**Expected behavior**
URL stems and templates should be moved to configuration files.

**Comment by tdonohue** [comment]
Agreed this is a minor issue. I'd welcome work in this area, but for now moving this to unscheduled backlog (until someone has time to fix one or more of these).

**Comment by mwoodiupui** [comment]
See another list of hard-coded identifier URLs at https://jira.lyrasis.org/browse/DS-4552
